### PR TITLE
endpoints: Allow all moz and chrome extensions to do CORS requests

### PR DESCRIPTION
**Pull request by johan-bjareholt**
Solves #42

### CODE REVIEW

**Comment by ErikBjare** [comment]
I don't like this, we should make sure the extension ID stays fixed, in both implementations. We do this for Thankful somehow, we should do the same for ActivityWatch.

Also, why isn't the Chrome extension included here?

https://github.com/ActivityWatch/aw-server/blob/c01ffd939b718c55152ffb81da55217ab8d799e6/aw_server/server.py#L71-L84

File: src/endpoints/cors.rs [hunk 1]
```diff
@@ -12,15 +12,10 @@ pub fn cors(config: &AWConfig) -> rocket_cors::Cors {
     if config.testing {
         allowed_exact_origins.push("http://127.0.0.1:27180".to_string());
     }
-    let mut allowed_regex_origins = vec![
-        "moz-extension://6b1794a0-5ae6-4443-aef9-7755717bb180".to_string(),
-        "moz-extension://02507908-8928-4ec3-86f5-2ae5dddf4c47".to_string(),  // The ID I got on Firefox for Android
```
Comment (ErikBjare) [inline src/endpoints/cors.rs]: If my permanent ID fix works, this Firefox line should no longer be needed.

**Comment by johan-bjareholt** [comment]
> Also, why isn't the Chrome extension included here? Is it even required for Chrome?

What do you mean it's not included? It's there in both on master and in this commit?

> Edit: I think I fixed it here ActivityWatch/aw-watcher-web#36

It refers to gecko, is some similar solution available for chrome or do you know if it is the same on all devices there?

**Comment by ErikBjare** [comment]
>What do you mean it's not included? It's there in both on master and in this commit?

I can't be blind, I see `cors_origins.append("moz-extension://*")` and `cors_origins.append("http://127.0.0.1:27180/*")` if testing but nothing else. There is **no** `cors_origins.append("chrome-extension://*")`.

>It refers to gecko, is some similar solution available for chrome or do you know if it is the same on all devices there?

It should be the same for all devices there, the Chrome Webstore assigns you a permanent one: https://stackoverflow.com/questions/8946325/chrome-extension-id-how-to-find-it

**Comment by johan-bjareholt** [comment]
> I can't be blind, I see cors_origins.append("moz-extension://*") and cors_origins.append("http://127.0.0.1:27180/*") if testing but nothing else. There is no cors_origins.append("chrome-extension://*").

That's odd, this is what I see

Master:

    let mut allowed_regex_origins = vec![
        "moz-extension://6b1794a0-5ae6-4443-aef9-7755717bb180".to_string(),
        "moz-extension://02507908-8928-4ec3-86f5-2ae5dddf4c47".to_string(),  // The ID I got on Firefox for Android
        "chrome-extension://nglaklhklhcoonedhgnpgddginnjdadi".to_string(),
    ];
    if config.testing {
        allowed_regex_origins.push("moz-extension://.*".to_string());
        allowed_regex_origins.push("chrome-extension://.*".to_string());
    }

This PR:

    let allowed_regex_origins = vec![
        "moz-extension://.*".to_string(),
        "chrome-extension://.*".to_string(),
    ];

> It should be the same for all devices there, the Chrome Webstore assigns you a permanent one: https://stackoverflow.com/questions/8946325/chrome-extension-id-how-to-find-it

Awesome, let's merge that PR then and I'll abandon this and then we'll add the final firefox id in aw-server-rust in a seperate PR or just push it straight to master.

**Comment by ErikBjare** [comment]
You were looking in aw-server-rust for the Chrome CORS origin. I was referring and linking to aw-server-python where it isn't needed for some reason...

**Pull request closed without merging**